tests/e2e-ui-tests/tests/mobile/workorder-workflow.spec.js

```diff
@@ -1,159 +1,107 @@
+        sessionStorage.clear();
+        localStorage.clear();
       });
+      authService.openMobileApp();
+      authService.verifyLoginPageIsVisible();
+      authService.loginToMobileApp(authData.users.TREVER.username,
+        authData.password.DEFAULT_PASSWORD);
+    });
 
-      step('select a workorder from the list', function () {
-        workorderService.selectWorkorderFromTheListByIndex(workorderData.mobileWorkorders.FIRST_WORKORDER.index);
-      });
+    step('verify workorders page is displayed', function() {
+      authService.verifySuccessfulLogin();
+    });
 
-      step('that workorder details are present and correct', function () {
-        workorderService.verifyWorkorderDetailsArePresentAndCorrect(workorderData.mobileWorkorders.FIRST_WORKORDER);
-      });
+    step('select a workorder from list', function() {
+      browser.sleep(20000);
+      workorderService.selectWorkorderFromTheList(workorderTitle);
+    });
 
-      step('that the begin workflow button is present', function () {
-        workorderService.verifyWorkorderWorkflowIsNotCompleted();
-      });
+    step('that workorder details are present and correct', function() {
+      browser.sleep(1000);
+      const workorderDetails = {
+        id: workorder.id,
+        status: 'New',
+        workorderName: workorderTitle
+      };
+      workorderService.verifyWorkorderDetailsArePresentAndCorrect(workorderDetails);
+    });
 
-      step('logout from mobile', function () {
-        authService.logoutOfMobileApp();
-      });
+    step('that the begin workflow button is present', function() {
+      workorderService.verifyWorkorderWorkflowIsNotCompleted();
     });
 
-    describe('SEARCH workorder tests', function () {
-      step('login to mobile as ' + authData.usernames.TREVER_SMITH, function () {
-        authService.openMobileApp();
-        authService.verifyLoginPageIsVisible();
-        authService.loginToMobileApp(authData.users.TREVER.username,
-          authData.password.DEFAULT_PASSWORD);
-      });
+    step('click on the begin workflow button', function() {
+      workorderService.beginWorkflow();
+    });
 
-      step('verify workorders page is displayed', function () {
-        authService.verifySuccessfulLogin();
-      });
+    step('verify elements of workflow completeion page are visible', function() {
+      workflowService.verifyWorkflowFormIsVisible();
+    });
 
-      step('search for a workorder in the list', function () {
-        workorderService.searchForWorkorderInList(workorderData.mobileWorkorders.lastWorkorder.workorderName);
-      });
+    step('click back button', function() {
+      workflowService.cancelWorkflowChanges();
+    });
 
-      step('verify there is only one workorder in the list', function () {
-        workorderService.verifyNumberOfWorkordersInList(1);
-      });
+    step('verify that the begin workflow button is still present', function() {
+      browser.sleep(1000);
```

Comment:
We shouldn't use a sleep value in tests. If you are waiting for an element to appear/disappear, then you can use the functions in `utils/wait.js`. 

This should appear inside the `verifyWorkorderWorkflowIsNotCompleted()` function of workorder service as the functions in the service object should contain all information associated with completing the tests.

Reply:
Created a jira for that https://issues.jboss.org/browse/RAINCATCH-1284